specification/walkthrough-1.md

```diff
@@ -0,0 +1,355 @@
+
+```{code-cell} python3
+---
+tags: ["remove_input"]
+---
+display_groups(inputs, ["run", "subject"])
+```
+
+### Subject level Node
+
+At this point, we have defined a `Model` that will be fit separate to each grouping-- in this case a separate time-series model for each `run`. 
+
+Next, we want to define a `subject` level node to pool together estimates from this `Node` for each `subject` using a fixed-effects model. 
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+:lines: 20-26
+```
+
+```{note}
+By default, `Nodes` are linked in sequential order, with all the `Contrast` outputs from a `Node` available to the subsequent `Node`.
+```
+
+#### From Run Outputs to Subject Inputs
+
+We need to use `GroupBy` to define how to group the outputs from the `Run` node
+as inputs to the `Subject` level:
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+:lines: 23
+```
+
+Here we are specifying that all images belonging to a single `subject` and from a single `contrast` should be grouped into a unit of analysis.
+
+Note that with 3 subjects and 2 runs, we will have 6 groups of output images from the `Run` node. Given two types of images (`variance` and `effect`), this results in 12 images that would be grouped as follows:
+
+```{code-cell} python3
+---
+tags: ["remove_input"]
+---
+outputs = pd.DataFrame.from_records(
+  [{
+     "subject": subject,
+     "run": run,
+     "contrast": contrast,
+     "image": f"sub-{subject}_task-simon_run-{run}_"
+              f"contrast-{contrast}_stat-{stat}_statmap.nii",
+   }
+   for subject in subjects for run in runs
+   for contrast in contrasts for stat in stats]
+)
+display_groups(outputs, ["subject", "contrast"])
+```
+
+```{tip}
+Although there is only one `contrast`, we include `contrast` as a grouping variable to be explicit.
+```
+
+#### Subject Model
+
+We can now specify the `Subject` level `Model`. 
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+:start-at: '"Model": {"X": [1], "Type": "meta"}'
+:lines: 1-2
+```
+
+Since our intent is to estimate the _mean effect_ for each subject, we only need an 
+intercept in our model. We specify the `"Type"` to be `Meta`, which is a 
+special type to identify fixed-effects models.
+
+```{note}
+`1` is a special variable to represent the intercept.
+```
+
+Remember that we must specify `Contrasts` in order to produce outputs for the
+next `Node`. `DummyContrasts` is a convenience function which will create contrasts with
+the weights `[1]` for all modeled inputs. Since we are not comparing anything at 
+the subject-level and simply want to pass forward the estimates generated by the 
+fixed-effects model, this is useful and saves us from specifying a more 
+verbose (but identical) `Contrast`.
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+:start-at: '"Model": {"X": [1], "Type": "meta"}'
+:lines: 1-2
+```
```

Comment:
This is the same as above. Did you want to compare to an explicit contrast?

Reply:
Good catch, I'll just remove this